docs/source/python/filesystems.rst

```diff
@@ -178,11 +179,31 @@ Example how you can read contents from a S3 bucket::
    >>> f.readall()
    b'some data'
 
+
+Note that it is important to configure :class:`S3FileSystem` with the correct
+region for the bucket being used. If `region` is not set, the AWS SDK will
+choose a value, defaulting to 'us-east-1' if the SDK version is <1.8.
+Otherwise it'll try to use a variety of heuristics to resolve the region.
```

Comment:
I would maybe add the "(environment variables, configuration profile, EC2 metadata server)" (copied from the C++ source), to make it clear that the "heuristics" don't include "infer it from the region"

```suggestion
Otherwise it will try to use a variety of heuristics to resolve the region.
```